specs/~dynamic-names.yml

```
@@ -0,0 +1,205 @@
+  Interpolation tag.
+
+  This tag's content refers to a key in the context whose value will be used in
+  place of the Dynamic Name itself as content of the tag. The name resolution is
+  identical to name resolution in Interpolation tags, as follows:
+    1) Split the name on periods; the first part is the name to resolve, any
+    remaining parts should be retained.
+    2) Walk the context stack from top to bottom, finding the first context
+    that is a) a hash containing the name as a key OR b) an object responding
+    to a method with the given name.
+    3) If the context is a hash, the data is the value associated with the
+    name.
+    4) If the context is an object, the data is the value returned by the
+    method with the given name.
+    5) If any name parts were retained in step 1, each should be resolved
+    against a context stack containing only the result from the former
+    resolution.  If any part fails resolution, the result should be considered
+    falsey, and should interpolate as the empty string.
+  The resolved data should be coerced into a string before being used as
+  content.
+
+  Set Delimiter tags MUST NOT affect the resolution of a Dynamic Name. The
+  Dynamic Names MUST be resolved against the context stack local to the tag.
+  Failed resolution of the dynamic name should result in nothing being rendered.
+
+  Engines that implement Dynamic Names MUST support their use in Partial tags.
+  In engines that also implement the optional inheritance spec, Dynamic Names
+  inside Parent tags should be supported as well. Dynamic Names cannot be
+  resolved more than once (Dynamic Names cannot be nested).
+tests:
+  - name: Basic Behavior - Partial
+    desc: The asterisk operator is used for dynamic partials.
+    data: { dynamic: 'content' }
+    template: '"{{>*dynamic}}"'
+    partials: { content: 'Hello, world!' }
+    expected: '"Hello, world!"'
+
+  - name: Context Misses - Partial
+    desc: Failed context lookups should be considered falsey.
+    data: { }
+    template: '"{{>*missing}}"'
+    partials: { missing: 'Hello, world!' }
+    expected: '""'
+
+  - name: Failed Lookup - Partial
+    desc: The empty string should be used when the named partial is not found.
+    data: { dynamic: 'content' }
+    template: '"{{>*dynamic}}"'
+    partials: { foobar: 'Hello, world!' }
+    expected: '""'
+
+  - name: Context
+    desc: The dynamic partial should operate within the current context.
+    data: { text: 'Hello, world!', example: 'partial'  }
+    template: '"{{>*example}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"*Hello, world!*"'
+
+  - name: Dotted Names
+    desc: The dynamic partial should operate within the current context.
+    data: { text: 'Hello, world!', foo: { bar: { baz: 'partial' } } }
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"*Hello, world!*"'
+
+  - name: Dotted Names - Failed Lookup
```

This one and the next have the same name. Maybe rename this one to "operator precedence".

```suggestion
  - name: Dotted Names - Operator Precedence
```